### PR TITLE
nixos/trippy: init

### PR DESCRIPTION
**Pull request by figsoda**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/trippy.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      enable = lib.mkEnableOption (lib.mdDoc "trippy");
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    security.wrappers.trip = {
+      owner = "root";
+      group = "root";
+      capabilities = "cap_net_raw+p";
+      source = "${pkgs.trippy}/bin/trip";
```
Comment (jian-lin) [inline nixos/modules/programs/trippy.nix:19]:
```suggestion
      source = lib.getExe pkgs.trippy;
```

File: nixos/modules/programs/trippy.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+in
+
+{
+  options = {
+    programs.trippy = {
+      enable = lib.mkEnableOption (lib.mdDoc "trippy");
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
```
Comment (jian-lin) [inline nixos/modules/programs/trippy.nix:14]: Also add it to `environment.systemPackages`?

File: nixos/modules/programs/trippy.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ config, lib, pkgs, ... }:
```
Comment (jian-lin) [inline nixos/modules/programs/trippy.nix:1]:
```suggestion
{ lib, config, pkgs, ... }:
```

https://nixos.org/manual/nixpkgs/unstable/#chap-conventions

> Arguments should be listed in the order they are used, with the exception of lib, which always goes first.

**Review by jian-lin — CHANGES_REQUESTED** [review]

File: nixos/modules/programs/trippy.nix (hunk 2 above)
Comment (figsoda) [reply nixos/modules/programs/trippy.nix:14]: We shouldn't need it since there is already a wrapper

**Review by jian-lin — APPROVED** [review]

**Pull request merged**